### PR TITLE
Activity Insight OA Workflow version checking job

### LINKED ISSUE #717: Implement file version check job in OA Workflow
We'll need to add a scope for files that need their version checked and implement the version checking job in the OAWorkflow task.  So any file that we have a record for, and we have downloaded the file, and has no version should be a part of this scope.

This is dependent on #575.

### PR DESCRIPTION
**Pull request by ajkiessl**
Branched off of https://github.com/psu-libraries/researcher-metadata/pull/713 .  

I added a `version_checked` boolean to `ActivityInsightOAFile` to be a status holder like `downloaded`.  This gets set to true when the job is queued so it doesn't get queued again.

I did my best to differentiate between the ScholarSphere deposit workflow job and this.  The name of this job ended up a little weird: `AiOAWfVersionCheckJob`

closes #717

### CODE REVIEW

File: app/models/activity_insight_oa_file.rb
Comment (EricDurante) [inline app/models/activity_insight_oa_file.rb:15]: oooh, I think that this scope is going to need to be changed per #797. I guess you could either do that here or make it a separate issue and wait until after this is merged.

cc @Smullz622

Comment (ajkiessl) [reply app/models/activity_insight_oa_file.rb:15]: Ah right.  I'll do that here.

File: app/services/oa_workflow_service.rb
Comment (EricDurante) [inline app/services/oa_workflow_service.rb:39]: Here again, I think it might be better to do 
```ruby
file.update_column(:version_checked, true)
```
instead of
```ruby
file.version_checked = true
file.save!
```

File: app/services/oa_workflow_service.rb
Comment (EricDurante) [inline app/services/oa_workflow_service.rb:44]: I think what we're doing here is ok since we're bypassing ActiveRecord and also re-raising the original error.

File: app/services/oa_workflow_service.rb
Comment (EricDurante) [inline app/services/oa_workflow_service.rb:42]: Should we add the `version_checked` attribute to the RailsAdmin list view of the files so that we can filter for the files that are in this state?

**Review by EricDurante — CHANGES_REQUESTED** [review]

File: app/jobs/ai_oa_wf_version_check_job.rb
Comment (ajkiessl) [reply app/jobs/ai_oa_wf_version_check_job.rb:20]: Yeah, the error handling here was added primarily to catch malformed PDF or wrong file type errors from the PDF parser.  I'll try to make the error handling more specific.

File: app/services/oa_workflow_service.rb
Comment (ajkiessl) [reply app/services/oa_workflow_service.rb:42]: Yeah, that is a good idea.  I'll do that.

File: app/jobs/ai_oa_wf_version_check_job.rb
Comment (ajkiessl) [reply app/jobs/ai_oa_wf_version_check_job.rb]: @EricDurante I moved the error handling into the version checker class and have it only rescuing a handful of PDF::Reader errors:
https://github.com/psu-libraries/researcher-metadata/blob/7647aa7f34fdb53a48dd0dafe52ce5a45ec3fb10/app/models/file_version_checker.rb#L50-L53

**Comment by ajkiessl** [comment]
@Smullz622 If you get a chance can I get a second set of eyes on the ActivityInsightOAFile scope tests and the OAWorkflowService tests?  I think I got everything in the merge.

**Review by Smullz622 — APPROVED** [review]
Looks good! Definitely think we need to start separating stuff out in some way though. Pretty sure I already have conflicts on my branch for 799

**Pull request merged**
